cmdtest.go

```diff
@@ -691,6 +691,25 @@ func fixedArgBuiltin(nargs int, f func([]string) ([]byte, error)) CommandFunc {
 	}
 }
 
+func rename(oldpath, newpath string) error {
+	// NOTE: This is used in place of os.Rename because it's
+	// able to rename files between different mount points.
+	src, err := os.Open(oldpath)
+	if err != nil {
+		return err
+	}
+	dst, err := os.Create(newpath)
+	if err != nil {
+		return err
+	}
+	defer dst.Close()
+	defer src.Close()
+	if _, err = io.Copy(dst, src); err != nil {
+		return err
```

Comment:
We need to let the user know here that things are corrupted: their original file is in an unknown state, but oldpath has the right contents.

(That's why I used the OS rename; it's basically atomic, on Unix at least.)

I would just add info to the error with fmt.Errorf.

Reply:
I'm thinking about something like the following:
```go
if n, err := io.Copy(dst, src); err != nil {
	return fmt.Errorf("go-cmdtest: could only copy %d bytes: %v", n, err)
}
```
What do you think, @jba?

Reply:
Mention that the destination file may be corrupted, too.